Code refactoring.

patrycja93 opened the pull request:
Moved input checking responsibility to InputChecker class. Deleted gameSettings in Match class and put there only UI and board size. Moved board displaying to Assessor class. Updated JudgeTest class
because there were changes in method responsibility for winner checking. Added new methods in the observable interface to communicate with spectators. Deleted PlayerChanger class because it was
unnecessary, Match class can keep information about active user and change active user.

patrycja93 committed:
Code refactoring.
Moved input checking responsibility to InputChecker class. Deleted gameSettings in Match class and put there only UI and board size. Moved board displaying to Assessor class. Updated JudgeTest class
because there were changes in method responsibility for winner checking. Added new methods in the observable interface to communicate with spectators. Deleted PlayerChanger class because it was
unnecessary, Match class can keep information about active user and change active user.

BulaMar commented on src/main/java/pl/patrycja/ox/Observable.java:
```diff
@@ -7,14 +7,18 @@
      * @param spectators is a list of objects which are spectators
      */
-    void inform(List<Spectator> spectators);
+    void informAboutPutSign(List<Spectator> spectators, int field, Sign sign);
```
`notifyThatSignWasPut` ? Current name sounds weird.
Please adjust other methods as well.

patrycja93 replied:
Ok I will change it.

BulaMar commented on src/main/java/pl/patrycja/ox/board/BoardExecutive.java:
```diff
@@ -59,15 +63,30 @@ public void clean() {
-    }
-}
+    }*/
+}
```
Please make changes to interface methods as discussed offline.

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Match.java:
```diff
@@ -1,68 +1,67 @@
-        gameSettings.getUi().display(board);
+        board.startMatch(spectators, numberOfMatch, activePlayer.getSign());
         while (!endMatch) {
-            turn();
```
That was not bad, maybe you can bring it back as `nextTurn()` with this while body?

patrycja93 replied:
Ok I will bring it back.

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Match.java:
```diff
@@ -1,68 +1,67 @@
 
-    @Override
-    public void inform(List<Spectator> spectators) {
+    public void checkIfMatchIsFinished(List<Spectator> spectators) {
```
👍

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Mode.java:
```diff
@@ -1,41 +1,54 @@
 
-    protected void checkIfCorrectInputData(String[] inputArrayParameters) {
+    void checkIfCorrectInputData(String[] inputArrayParameters) {
+        InputChecker inputChecker = new InputChecker(ui);
```
Maybe this could be in UI?

patrycja93 replied:
Yes, I will handle it in the next pull request.

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Mode.java:
```diff
@@ -1,41 +1,54 @@
+    Player askWhichPlayerStarts(List<Player> players) {
         InputChecker inputChecker = new InputChecker(ui);
-        inputChecker.checkIfInputParametersAreValid(inputArrayParameters);
+        ui.display("Which player should start: O or X ? ");
```
Make sens only for now as you have only two possible signs. If there are no requirements for now then ignore also for now.

BulaMar commented on src/main/java/pl/patrycja/ox/gameflow/Mode.java:
```diff
@@ -1,41 +1,54 @@
-        inputChecker.checkIfInputParametersAreValid(inputArrayParameters);
+        ui.display("Which player should start: O or X ? ");
+        Sign sign = inputChecker.checkSignValidate(ui.read());
+        //noinspection OptionalGetWithoutIsPresent player always be there
```
Similar here, if there will be more signs this will not be true, can be skipped for now. Discussed offline.

BulaMar commented on src/main/java/pl/patrycja/ox/ui/ConsoleUI.java:
```diff
@@ -19,4 +20,24 @@ public void display(Object input) {
+        return Integer.parseInt(next);
+    }
+
+    private boolean invalid(String next) {
```
Not used.

BulaMar commented on src/main/java/pl/patrycja/ox/ui/InputChecker.java:
```diff
@@ -21,14 +28,14 @@ public InputChecker(UI ui) {
         return Integer.parseInt(String.valueOf(input));
     }
 
-    public void checkIfInputParametersAreValid(String[] parameters) {
```
That was fine.

BulaMar commented on src/main/java/pl/patrycja/ox/ui/InputChecker.java:
```diff
@@ -39,12 +46,23 @@ public void checkIfInputParametersAreValid(String[] parameters) {
     }
 
-    private boolean checkIfInteger(String number, UI ui) {
+    public void checkIfUnbrokenLineIsMoreThanBoardSize(String[] parameters) {
```
`checkIfUnbrokenLineIsGraterThanBoardSize`

BulaMar commented on src/main/java/pl/patrycja/ox/ui/InputChecker.java:
```diff
@@ -39,12 +46,23 @@ public void checkIfInputParametersAreValid(String[] parameters) {
+        if (parameters.length > 1 && (Integer.parseInt(parameters[0]) < Integer.parseInt(parameters[1]))) {
+            int defaultValue = 3;
+            String min = String.valueOf(Math.max(Integer.parseInt(parameters[0]), defaultValue));
+            String max = String.valueOf(Math.max(Integer.parseInt(parameters[1]), defaultValue));
```
Please give more meaningful names to min and max

BulaMar commented on src/main/java/pl/patrycja/ox/ui/InputChecker.java:
```diff
@@ -62,4 +80,12 @@ private boolean checkIfNumberIsInRange(String number, int boardSize, UI ui) {
         }
     }
+
+    public Sign checkSignValidate(String sign) {
```
validateSign or checkSignValidity

BulaMar reviewed (changes requested):
Nice! 👍 Please make small changes, mostly those discussed offline.

patrycja93 committed:
After code review.
Changed sign to player in spectator's methods to have the possibility display players name and scores in the future. Corrected little grammatic errors in methods name and variables.

BulaMar commented on src/main/java/pl/patrycja/ox/winnerchecker/Assessor.java:
```diff
@@ -41,11 +44,16 @@ public boolean isMatchOver() {
 
+    @Override
+    public void playerHasChanged(Player player) {
+
```
Should it be empty? Please put at least comment or TODO.
If this will stay empty then maybe you should rethink Spectator interface and usage of it.

BulaMar reviewed (approved):
Thanks for changes 🙇  
one last small remark.

patrycja93 committed:
Added todo to empty method to remember implement it.
